src/main/java/util/RetryHelper.java

```diff
@@ -0,0 +1,34 @@
+package util;
+
+import java.util.function.Supplier;
+
+public final class RetryHelper {
+    private final int maxRetryCount;
+    private int retryCount;
+
+    public RetryHelper(int maxRetryCount) {
+        this.maxRetryCount = maxRetryCount;
+    }
+
+    public <E> E retry(Supplier<E> supplier) {
+        E result = null;
+        while (result == null && retryCount <= maxRetryCount) {
+            result = useSupplier(supplier);
+            retryCount += 1;
+        }
+        if (result == null) {
+            throw new RuntimeException("프로그램이 비정상 종료되었습니다.");
+        }
+        return result;
+    }
+
+    private <E> E useSupplier(Supplier<E> supplier) {
+        try {
+            return supplier.get();
+        } catch (RuntimeException e) {
+            System.out.println(e.getMessage());
+        }
+        return null;
+    }
```

Comment:
[제안] 해당 부분은 지극히 개인적인 제안이니 참고만 해주세요 :) 

null을 return하는 부분이 catch 문 밖에 있어야 할 이유가 없어보였어요. try 문에서 early return을 해준다면, null을 return하는 케이스는 catch 문을 통과한 케이스 뿐입니다. 그렇다면 null을 return하는 것은 catch 문에서 행해지는 행위로 보여지고 동일한 맥락에서 진행되어야 하는 코드가 모여있다면 코드를 읽는 사람으로 하여금 더 좋은 가독성을 제공해 줄 수 있다고 생각합니다.

```suggestion
    private <E> E useSupplier(Supplier<E> supplier) {
        try {
            return supplier.get();
        } catch (RuntimeException e) {
            System.out.println(e.getMessage());
            return null;
        }
    }
```